conference.js

```diff
@@ -783,6 +787,13 @@ export default {
      * dialogs in case of media permissions error.
      */
     muteAudio(mute, showUI = true) {
+        if (!mute
+                && isUserInteractionRequiredForUnmute(APP.store.getState())) {
```

Comment:
I think this check is redundant because we have the check in the middleware, isn't it? 

If we want to be extra protected shouldn't we add this for all unmute operations in conference.js? Or maybe move it in `muteLocalAudio`?

Reply:
The check I'd say is not redundant because this logic does logic that fires before middleware. I'd say the logic entire function is redundant, but I'm not going to attempt that fix right now.

Reply:
To be honest, this is the place that's probably more important for the fix due to how the external api works. The checks in middleware were added for safety but maybe those middleware checks should be removed. Let me know.

Reply:
Maybe I got very confused, please tell me where is the mistake. This is how I see it in the code:
muteAudio -> muteLocalAudio -> dispatch(setAudioMuted(muted)) -> the middleware -> mutes the track effectively

So I think your middleware check will stop the muting anyway, isn't it?

If you remove the middleware check, we have a few other paths that will effectively unmute the track, they will again go trough the middleware but not trough `muteAudio`. For example the unmute button of the "speaking while muted" notification which calls directly `muteLocalAudio`.

Reply:
The problem path is: conference#muteAudio -> createLocalTracksF -> conference#useAudioStream.

Reply:
Oh ok. Now I understand!

So I guess both checks should stay.